Fade long channel names in sidebar

maxpoulin64 opened the pull request:
This uses a few CSS tricks to fade away long channel names in the sidebar so the badge don't overlap on top of the channel names.

xPaw commented on client/css/style.css:
```diff
@@ -272,12 +272,19 @@ button, .user {
+	position: absolute; left: 10px;
+}
+#sidebar .chan:after {
+	position: absolute; right: 10px; top: 0; bottom: 0; z-index: 0;
```
One property per line please.

astorije commented:
Screenshots?

xPaw commented:
Okay, I tried this now, and there's one thing I don't like about it. Even if you delete close button element, the name still doesn't expand past the notification badge. It simply cuts off where the notification badge would be, without any gradient.

If you can fix this, that would be nice!

maxpoulin64 commented:
Updated. I don't know where exactly you saw that, but I changed the place where the gradient is applied so the trick is simpler which makes it easier to make look clean. I also made sure to move the badge out of the way by putting it back to position:absolute when it fades away. It should look pretty good now no matter what.

![lounge-badge](https://cloud.githubusercontent.com/assets/5481612/13200387/32b6ba7a-d813-11e5-8db3-3a985f60b5fd.gif)

xPaw commented:
:+1:

YaManicKill commented:
This is pretty awesome, thanks @maxpoulin64 

Can you rebase, and check that the new stylelint doesn't break the build for this, and then I'm happy to merge. 

:+1:

maxpoulin64 commented:
@YaManicKill Here you go. It did break all over the place, and it didn't merge properly at all either. Ended up just redoing it over master.

astorije commented:
I love it, excellent job @maxpoulin64!

I did notice a small issue though: when the name is long, hovering on the name is messing up with the channel closing button (&times;). I know we mentioned removing it, but it's not the case yet (and it is not sure yet). Here is what I'm talking about (notice the first and third lines):

![thelounge-fade-badge](https://cloud.githubusercontent.com/assets/113730/13210509/3071138c-d8fe-11e5-9c87-28964d6ca1ca.gif)

Do you think you could fix this before we merge? :-/

astorije commented:
After discussing with @maxpoulin64 on IRC, he fixed this and it does look terrific. So :+1: and merging.

Also, I'm re-labeling this as a bugfix since currently long channel names are simply not being displayed well: they appear on a second line and get cut at the end of the end of the sidebar. Leaving as enhancement too considering @maxpoulin64 went well beyond duty :-)
